spec/solver_spec.rb

```diff
@@ -0,0 +1,23 @@
+require_relative '../solver.rb'
+
+describe Solver do
+    
+    before(:all) do
+        @solver = Solver.new
+    end
+
+    it 'test factorial method' do
+        @solver.factorial(2) == 2
+        @solver.factorial(3) == 6
+        @solver.factorial(10) == 3628800
+    end
+    it 'test reverse method' do
+        @solver.reverse('hello') == 'olleh'
+    end
+    it 'test factorial method' do
+        @solver.fizzbuzz(20) == 'fizzbuzz'
+        @solver.fizzbuzz(10) == 'buzz'
+        @solver.fizzbuzz(6) == 'fizz'
+        @solver.fizzbuzz(7) == '7'
```

Comment:
- [ ] kindly note that When N is divisible by 3 and 5, the function should return "fizzbuzz". Could you please correct the function or the test?
- [ ] Kindly use the `expect` syntax to test each test case, Example 👇 
```ruby
 expect(solver.fizzbuzz(3)).to eql('fizz')  # N is divisible by 3
    ...
    expect(solver.fizzbuzz(5)).to eql('buzz')  # N is divisible by 5
    ...
    expect(solver.fizzbuzz(15)).to eql('fizzbuzz') # N is divisible by 3 and 5
    ...
    expect(solver.fizzbuzz(7)).to eql('7')  # N is not divisible by 3 or 5
```